File: c++ version/1-100/83-remove-duplicates-from-sorted-list.cpp
ListNode* deleteDuplicates(ListNode* head) {
    if (head == nullptr) return head;
    ListNode *prev = head;
    ListNode *curr = prev->next;
    while (curr) {
        if (prev->val != curr->val) {
            prev = curr;
            curr = curr->next;
        }
        else {
            prev->next = curr->next;
            ListNode *tmp = curr;
            curr = prev->next;
            delete tmp;
        }
    }
    return head;
}

File: python version/83-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head: return None
        prev = head
        nxt = head
        while nxt:
            if nxt.val == prev.val:
                nxt = nxt.next
            else:
                prev.next = nxt
                prev = prev.next
                nxt = nxt.next
        prev.next = nxt
        
        return head